internal/health_checks/health_check_error.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health_checks
+
+type HealthCheckError struct {
+	Code         string
+	Class        string
+	Message      string
+	Action       string
+	ResourceLink string
+	IsFatal      bool
+	Err          error
+}
+
+func (e HealthCheckError) Error() string {
+	return e.Err.Error()
+}
+
+var (
+	FB_METRICS_PORT_ERR = HealthCheckError{
+		Code:         "FB_METRICS_PORT_ERR",
+		Class:        "PORT",
+		Message:      "Port 20202 used for fluent-bit self metrics is unavailable.",
+		Action:       "Verify the 20202 port is available.",
```

Comment:
Suggested change addressed in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/d2845edbb605e75924e9a00dacde743bd776558b . 

Summary and details :
- Maintaining commands to verify ports (or any other simple command check) is complicated to maintain in different platforms. For this specific case we could do here, `netstat -na` for both Windows (powershell) and Linux (if there is not a weird distro we support that doesn't install netstat by default). We would need to add a way (a constructor or something) to convert the `HealthCheckError` to have a platform specific error. Then we would be able to have something like `netstat -na | Select-String "20202"` for Windows and `netstat -na | grep "20202"` for Linux. WDYT ? CC @jinghan-ma

- I will try to take most of the specificity for the doc links. They are I think in a good state, but having a nicer `troubleshooting page` and also improving the `Authorize the Ops Agent` page would make big difference too.

- Having a documentation page that serves the purpose of solving this specific errors, so it is updated in sync with this would be the best I believe (so it doesn't change latter and get 404 errors).

- The current resource links are drawn from the best I believe that can give pointers to solve the issue.  In case of adding the correct scopes to the VM, there is no page that has steps for that :( .

Reply:
I suppose we don't have to solve it in this PR; my goal is to make sure that we're giving users sufficient direction to self-diagnose. As long as we're linking to something relevant then we can incrementally work on that over time without blocking this PR.

Ideally much of this will be automatable in the future anyhow, e.g. if the agent can't bind a port then we could extend the port check to run `netstat -na` automatically to attempt to diagnose the problem -- and then we can return a message like "Port 20202 needed by the Ops Agent is currently in use by <process name>" instead of just "Port is unavailable". But again these can all be worked on incrementally after the first cut.